packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -425,7 +425,35 @@ class Scaffold extends StatefulWidget {
   ///   );
   /// }
   /// ```
-  static ScaffoldState of(BuildContext context) => context.ancestorStateOfType(const TypeMatcher<ScaffoldState>());
+  ///
+  /// If there is no [Scaffold] in scope, then this will throw an exception.
+  /// To return null if there is no [Scaffold], then pass `nullOk: true`.
+  static ScaffoldState of(BuildContext context, { bool nullOk: false }) {
+    assert(nullOk != null);
+    assert(context != null);
+    ScaffoldState result = context.ancestorStateOfType(const TypeMatcher<ScaffoldState>());
+    if (nullOk || result != null)
+      return result;
+    throw new FlutterError(
+      'Scaffold.of() called with a context that does not contain a Scaffold.\n'
+      'No Scaffold could be found starting from the context that was passed to Scaffold.of(). '
```

Comment:
(Will fix in subsequent patch, since moving a branch to another development machine is a pain.)